client/src/featureform/register.py

```diff
@@ -802,9 +849,19 @@ def __call__(self, fn):
             if self.name is nv[0] and self.variant is nv[1]:
                 raise ValueError(f"Transformation cannot be input for itself: {self.name} {self.variant}")
         self.query = dill.dumps(fn.__code__)
-        fn.register_resources = self.register_resources
-        fn.name_variant = self.name_variant
-        return fn
+        # Given the function type does not implement __getitem__ we need to wrap it
```

Comment:
Noticed that you copy and pasted this, would it make sense to make it its own function or put the logic in the constructor?

Reply:
Definitely cleaner to wrap this up in the wrapper class. Thanks for the suggestion!